src/Symfony/Component/ExpressionLanguage/SyntaxError.php

```diff
@@ -13,8 +13,14 @@
 
 class SyntaxError extends \LogicException
 {
-    public function __construct($message, $cursor = 0)
+    public function __construct($message, $cursor = 0, $expression = '')
     {
-        parent::__construct(sprintf('%s around position %d.', $message, $cursor));
+        $message = sprintf('%s around position %d', $message, $cursor);
+        if ($expression) {
+            $message = sprintf('%s for expression "%s"', $message, $expression);
```

Comment:
I suggest using backticks rather than quotes to encore the expression, because quotes can appear inside the expression itself.